### ISSUE TITLE
cistem: contents of the archive look like HTML

### ISSUE THREAD

**Issue by des2037** [spack/spack#7576]
I'm trying to install cistem, but it Spack states that the archive looks like HTML:

### Actual Result
```
==> Installing cistem
==> Fetching https://cistem.org/system/tdf/upload3/cistem-1.0.0.tar.gz?file=1&type=cistem_details&id=37&force=0

==> Warning: The contents of 
  the archive
   look like HTML.
  The checksum will likely be bad.  If it is, you can use
  'spack clean <package>' to remove the bad archive, then
  fix your internet gateway issue and install again.
==> Error: OSError: [Errno 2] No such file or directory
`OSError: OSError: [Errno 2] No such file or directory

./spack/lib/spack/spack/package.py:1019, in do_fetch:
     25                                     self.spec.format('$_$@'), ck_msg)
     26    
     27            self.stage.create()
  >> 28            self.stage.fetch(mirror_only)
     29            self._fetch_time = time.time() - start_time
     30    
     31            if spack.do_checksum and self.version in self.versions:`
```

I think the problem is the download link. If I grab the file with wget:

`wget "https://cistem.org/system/tdf/upload3/cistem-1.0.0.tar.gz?file=1&type=cistem_details&id=37&force=0"`

Then the delivered file has the name: 
`cistem-1.0.0-beta-source-code.tar.gz?file=1&type=cistem_details&id=37&force=0`

If I truncate the link to:
`https://cistem.org/system/tdf/upload3/cistem-1.0.0.tar.gz`

I just get an empty file when I try to download it. So I think the problem may be that the archive doesn't end in .tar.gz, and therefore Spack doesn't know what to do with it.

Thanks in advance for any help! 

### Steps to reproduce the issue

```console
$ spack install cistem
```

### Information on your system

CentOS Linux release 7.4.1708 (Core)

**Comment by adamjstewart** [comment]
I believe your diagnosis is correct. When I run `spack fetch cistem`, the error message I get is:
```
==> Warning: The contents of 
  the archive
   look like HTML.
  The checksum will likely be bad.  If it is, you can use
  'spack clean <package>' to remove the bad archive, then
  fix your internet gateway issue and install again.
==> Error: md5 checksum failed for /Users/Adam/spack/var/spack/stage/cistem-1.0.0-2dk2torxauuiaizogtnuy7dzlkttz3bm/cistem-1.0.0.tar.gz?file=1&type=cistem_details&id=37&force=0
Expected 479f395b30ad630df3cbba9c56eb29c2 but got d41d8cd98f00b204e9800998ecf8427e
```
If I download it manually, I get the correct checksum. But if I truncate everything after the `.tar.gz`, I get the same checksum that Spack is getting. For some reason Spack is truncating everything after the extension when fetching. @tgamblin and @scheibelp have the most commits to the fetcher so they may know more than I do.

**Comment by des2037** [comment]
Thanks for looking into this Adam, much appreciated! I'll investigate:

`./spack/lib/spack/spack/url.py`
`./spack/lib/spack/spack/fetch_strategy.py`

to see what I can discover, though my I'm pretty much a python novice, lol.

**Comment by alalazo** [comment]
This seems to be fixed at 5eca4f14706a9590407f14c9c76e5a8bb663e9df:
```console
$ spack fetch cistem
[ ... ]
==> Fetching https://cistem.org/system/tdf/upload3/cistem-1.0.0-beta-source-code.tar.gz?file=1&type=cistem_details&id=37&force=0
########################################################################################################################################### 100,0%
```
